signal/signals_android.c

```diff
@@ -0,0 +1,153 @@
+// ======================================================================================
```

Comment:
Does it miss `// +build library,android`?

Reply:
Uhm, I don't think so, but it might be worth adding. The main issue here is that both build tags and file naming conventions are kinda redundant, and I don't want to bring confusion by using both. So, here I used `_android.c` name and it seems to work properly because CGo now is enabled (and this means that C files are even considered by compiler to look into) only on `library` builds (and that, in its turn enforced by build tags in .go files).

So it's a bit complicated, but I'm going to refactor signals anyway, so at least this combination works for now. (I need to address tests fails on CI though, maybe they're related)

Reply:
So does it mean that naming a file `signals_android.c` will compile it only for `android` environment?

Reply:
@adambabik yes! I didn't find any mention of it in docs, but empirically that's the way how it seemed to work (I spent almost a day, trying out those things and waiting for xgo builds)

Reply:
Thanks, now it makes more sense!